src/main/java/org/deepsymmetry/beatlink/AnnouncementSocketConnection.java

```diff
@@ -0,0 +1,294 @@
+package org.deepsymmetry.beatlink;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.*;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class AnnouncementSocketConnection extends LifecycleParticipant implements SocketSender {
+    private static final Logger logger = LoggerFactory.getLogger(AnnouncementSocketConnection.class);
+
+    public static final AnnouncementSocketConnection connectionsManager = new AnnouncementSocketConnection();
+
+    /**
+     * The socket used to receive device status packets while we are active.
+     */
+    private final AtomicReference<DatagramSocket> announcementSocket = new AtomicReference<DatagramSocket>();
+
+    /**
+     * Track when we started listening for announcement packets.
+     */
+    private static final AtomicLong startTime = new AtomicLong();
+
+    /**
+     * Maintain a set of addresses from which device announcements should be ignored. The {@link VirtualCdj} will add
+     * its socket to this set when it is active so that it does not show up in the set of devices found on the network.
+     */
+    private final Set<InetAddress> ignoredAddresses =
+            Collections.newSetFromMap(new ConcurrentHashMap<InetAddress, Boolean>());
+
+    /**
+     * The port to which devices broadcast announcement messages to report their presence on the network.
+     */
+    public static final int ANNOUNCEMENT_PORT = 50000;
+
+    /**
+     * Check whether we are presently posing as a virtual CDJ and receiving device status updates.
+     *
+     * @return true if our socket is open, sending presence announcements, and receiving status packets
+     */
+    public boolean isRunning() {
+        return announcementSocket.get() != null;
+    }
+    public static AnnouncementSocketConnection getInstance() {
+        return connectionsManager;
+    }
+
+    /**
+     * Send a master changed announcement to all registered master listeners.
+     *
+     * @param announcement the message announcing the new announcements
+     */
+    private void deliverDeviceAnnouncement(final DeviceAnnouncement announcement) {
+        for (final AnnouncementListener listener : getDeviceAnnouncementListeners()) {
+            try {
+                listener.handleDeviceAnnouncement(announcement);
+            } catch (Throwable t) {
+                logger.warn("Problem delivering announcement to listener", t);
+            }
+        }
+    }
+
+    /**
+     * Start listening for device announcements and keeping track of the DJ Link devices visible on the network.
```

Comment:
It looks like you’re leaving this responsibility (keeping track of the devices we currently believe exist on the network) in `DeviceFinder`, which makes sense, but this comment should be updated then.